src/py/tests/test_page_generator.py

```diff
@@ -155,37 +189,256 @@ async def test_page_generator():
-    assert with_others_result_re.findall(with_others), (
-        f"{len(with_others_result_raw)}: {with_others_result_raw}"
-        "\n"
-        f"{len(with_others)}: {with_others}"
-    )
+    scripts, encodings = get_scripts_from_html(combined)
+
+    # Calculate expected count
+    expected_count = 0
+    script_index = 0
+
+    # Mathjax adds one (unless False)
+    if custom_mathjax is not False:
+        expected_count += 1
+        if custom_mathjax is None:
+            expected_mathjax = DEFAULT_MATHJAX
+        elif isinstance(custom_mathjax, tuple):
+            expected_mathjax = str(custom_mathjax[0])
+            assert encodings[script_index] == custom_mathjax[1]
+        else:
+            expected_mathjax = str(custom_mathjax)
+        assert scripts[script_index] == expected_mathjax
+        script_index += 1
+
+    # Plotly always adds one
+    expected_count += 1
+    if isinstance(custom_plotly, tuple):
+        assert scripts[script_index] == str(custom_plotly[0])
+        assert encodings[script_index] == custom_plotly[1]
+    else:
+        assert scripts[script_index] == str(custom_plotly)
+    script_index += 1
+
+    # Others adds however many are in the list
+    expected_count += len(other_scripts)
+    for script in other_scripts:
+        if isinstance(script, tuple):
+            assert scripts[script_index] == str(script[0])
+            assert encodings[script_index] == script[1]
+        else:
+            assert scripts[script_index] == str(script)
+        script_index += 1
+
+    # Kaleido scopes always adds one
+    expected_count += 1
+    assert scripts[script_index].endswith("kaleido_scopes.js")
 
+    assert len(scripts) == expected_count
+
+
+# Test file path validation
+async def test_existing_file_path(temp_js_file):
+    """Test that existing file paths work with and without file:/// protocol."""
+    # Test with regular path
+    generator = PageGenerator(plotly=str(temp_js_file))
+    html = generator.generate_index()
+    scripts, encodings = get_scripts_from_html(html)
+    assert len(scripts) == 3  # noqa: PLR2004
+    assert scripts[0] == DEFAULT_MATHJAX
+    assert scripts[1] == str(temp_js_file)
+    assert scripts[2].endswith("kaleido_scopes.js")
+
+    # Test with file:/// protocol
+    generator_uri = PageGenerator(plotly=temp_js_file.as_uri())
+    html_uri = generator_uri.generate_index()
+    scripts_uri, encodings_uri = get_scripts_from_html(html_uri)
+    assert len(scripts_uri) == 3  # noqa: PLR2004
+    assert scripts_uri[0] == DEFAULT_MATHJAX
+    assert scripts_uri[1] == temp_js_file.as_uri()
+    assert scripts_uri[2].endswith("kaleido_scopes.js")
+
+
+async def test_nonexistent_file_path_raises_error(
+    nonexistent_file_path,
+    nonexistent_file_uri,
+):
+    """Test that nonexistent file paths raise FileNotFoundError."""
+    # Test with regular path
+    with pytest.raises(FileNotFoundError):
+        PageGenerator(plotly=str(nonexistent_file_path))
```

Comment:
Should test the case where a `Path` is passed as well, right? For this test and the others.

```suggestion
    # Test with Path object
    with pytest.raises(FileNotFoundError):
        PageGenerator(plotly=nonexistent_file_path)
        
    # Test with path as string
    with pytest.raises(FileNotFoundError):
        PageGenerator(plotly=str(nonexistent_file_path))
```